### ISSUE TITLE
Direct chat missing in Spaces

### ISSUE THREAD

**Issue by Thatoo** [vector-im/element-web#19106]
### Steps to reproduce

In a Space, not all direct chat appear but they do on Android Element.

### What did you expect?
Have all direct chat belonging to a space being listed
### What happened?
They don't appear. Need to go on Home space...

### Operating system

Debian BUster

### Application version

1.8.5

### How did you install the app?

from website element.io

### Homeserver

matrix.leprette.fr or leprette.fr (it's confusing)

### Have you submitted a rageshake?

Yes

**Comment by Thatoo** [comment]
I'm opening a new issue because this one has been closed whereas the problem isn't solved, for me at least : https://github.com/vector-im/element-web/issues/17291

**Comment by Thatoo** [comment]
Of course, if I can be of any help to test rooms and see identify the reason why it is these rooms that are missing in spaces, I'd be glad to help.

**Comment by SimonBrandner** [comment]
@Thatoo, could you please test if you see this on staging.element.io? And could you please check the id of the room (it can be found in `Advanced` room settings) is in `/devtools` -> `Explore Account Data` -> `m.direct`?

**Comment by Thatoo** [comment]
The problem exist also on staging.element.io .

The id of the room is !unigLiZUuDtVClghjY:synapse.leprette.fr which is funny because synapse.leprette.fr doesn't exist anymore. 
Our server is matrix.leprette.fr now. I guess the id haven't been updated since the creation of the room.

If it helps to understand the problem and help to solve it, here is a little history of this room :
Originaly, members in this direct chat were @AA:synapse.leprette.fr and @BB:synapse.leprette.fr then when we created our new server matrix.leprette.fr. We invited to this direct chat (it became a room then...) the new users @AA:leprette.fr and @BB:leprette.fr (for our matrix id on this new server, we use leprette.fr instead of matrix.leprette.fr) and our old users @AA:synapse.leprette.fr and @BB:synapse.leprette.fr left the room. Then I had to login with an old version of Riot (I can't remeber which one) to switch all these room to direct chat(the option had disappear with new version of Riot and then Element, is it back now...?).
We did that for all our rooms and when everything were ok, we switched off our old server (it is now long gone...)
Everything was working just fine till now.
It's only because I face this problem that I remember this history of our server.
Maybe I should check if the other member of this room face the same issue on Space.
By the way, I realise that some rooms are still in version 1. Is it important to upgrade version of rooms? Is it possible to upgrade all room at once?

**Comment by t3chguy** [comment]
Check your account_data `m.direct` event for `!unigLiZUuDtVClghjY:synapse.leprette.fr` and which user it is listed as a DM with, if at all.

**Comment by Thatoo** [comment]
The old user @aa:synapse.leprette.fr is listed but not the new user @aa:leprette.fr.
How can I make the replacement?

**Comment by Thatoo** [comment]
The problem seems to have been solved thanks to `/converttodm`

**Comment by SimonBrandner** [comment]
In that case, I think we can close this, let us know if you ever see this again
